webhook/resources/backingimage/mutator.go

```diff
@@ -35,14 +41,61 @@ func (b *backingImageMutator) Resource() admission.Resource {
 }
 
 func (b *backingImageMutator) Create(request *admission.Request, newObj runtime.Object) (admission.PatchOps, error) {
-	return mutateBackingImage(newObj)
-}
+	var patchOps admission.PatchOps
 
-func (b *backingImageMutator) Update(request *admission.Request, oldObj runtime.Object, newObj runtime.Object) (admission.PatchOps, error) {
-	return mutateBackingImage(newObj)
+	backingImage := newObj.(*longhorn.BackingImage)
+
+	name := util.AutoCorrectName(backingImage.Name, datastore.NameMaximumLength)
+	if name != backingImage.Name {
+		patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/metadata/name", "value": "%s"}`, name))
+	}
+
+	checksum := strings.TrimSpace(backingImage.Spec.Checksum)
+	patchOps = append(patchOps, fmt.Sprintf(`{"op": "replace", "path": "/spec/checksum", "value": "%s"}`, checksum))
+
+	// Handle Spec.SourceParameters
+	parameters := make(map[string]string, 0)
+	for k, v := range backingImage.Spec.SourceParameters {
+		parameters[k] = strings.TrimSpace(v)
+	}
+
+	if longhorn.BackingImageDataSourceType(backingImage.Spec.SourceType) == longhorn.BackingImageDataSourceTypeExportFromVolume {
```

Comment:
BTW, could we add enum schema validation for `SourceType`?

Reply:
Yep, we should add `enum` for these data types.